test/corpus/forloop.txt

```diff
@@ -0,0 +1,78 @@
+======================
+For loop in array
+======================
+
+[
+  i
+  for i in std.range(0,10)
+]
+
+---
+
+(document
+  (expr
+    (expr (id))
+    (forspec
+      (id)
+      (expr (expr (expr (id)) (id))
+      (args (expr (number)) (expr (number)))))))
+
+======================
+For loop in object
+======================
+
+{
+  ['item' + i]: i
+  for i in std.range(0,10)
+}
+
+---
+
+(document
+  (expr
+    (field
+      (fieldname
+        (expr
+          (expr
+              (string
+                (string_start)
+                (string_content)
+                (string_end)))
+          (expr (id))))
+      (expr (id)))
+      (forspec
+        (id)
+        (expr (expr (expr (id)) (id))
+        (args (expr (number)) (expr (number)))))))
+
+================================
+For loop in object with function
+================================
+
+{
+  ['item' + i](v): i + v
+  for i in std.range(0,10)
+}
```

Comment:
This isn't possible according to the specification but is correctly intepreted by jsonnet.

```jsonnet
// a.jsonnet
local lib = {
  ['f' + i](a): i + a
  for i in std.range(0, 10)
};

lib.f10(3)
```

```console
[I] ➜ jsonnet a.jsonnet
13

[I] ➜ cjsonnet a.jsonnet
13

[I] ➜ jrsonnet a.jsonnet
13

```